paginate.md

```diff
@@ -0,0 +1,167 @@
+# Adding Pagination
+
+Using the pagination library, [will_paginate](https://github.com/mislav/will_paginate), we will add pagination to our `books` `index` view so we don't create an infinite scrolling scenario.
+
+## Step 1 - Installing will_paginate
+
+To start using [will_paginate](https://github.com/mislav/will_paginate), first add it to your `Gemfile`: 
+```rb
+# bnb-library/Gemfile
+---------------------
+
+gem 'trestle-auth'
+gem 'faker', :git => 'https://github.com/faker-ruby/faker.git', :branch => 'master'
+gem 'will_paginate'
+```
+Bundle your Gemfile: 
+```
+dip bundle install
+```
+
+## Step 2 - Pagination and Model / View Integration
+
+[will_paginate](https://github.com/mislav/will_paginate) makes creating a pagination query extremely simple. All we to do is update our `books_controller` `index` view and action. 
+
+Open the file: 
+```rb
+# app/controllers/books_controller.rb
+-------------------------------------
+
+class BooksController < ApplicationController
+  before_action :set_book, only: %i[ show edit update destroy ]
+
+  # GET /books or /books.json
+  def index
+    @books = Book.all
+  end
+
+.. .
+```
+Instead of querying for `Book.all` we are going to instead utilize the `.paginate` method included in the [will_paginate](https://github.com/mislav/will_paginate) gem. It takes a paramter of `:page` that will recieve its value, and let [will_paginate](https://github.com/mislav/will_paginate) know what page to display when you (the user) toggle between different page links.
+```rb
+# app/controllers/books_controller.rb
+-------------------------------------
+
+class BooksController < ApplicationController
+  before_action :set_book, only: %i[ show edit update destroy ]
+
+  # GET /books or /books.json
+  def index
+    @books = Book.paginate(page: params[:page], per_page: 10)
+  end
+
+.. .
+```
+In order to see the pagination links we must update our `index` view.
+
+Open the file: 
+
+```haml
+# app/views/books/index.html.haml
+---------------------------------
+
+%h1 Books
+%table
+  %thead
+    %tr
+      %th Title
+      %th Description
+      %th Price
+      %th{:colspan => "3"}
+  %tbody
+    - @books.each do |book|
+      %tr
+        %td= book.title
+        %td= book.description
+        %td= book.price
+        %td= button_to 'Show', book_path(id: book.id), method: :get
+%br/
+= link_to 'Home', home_index_path, class: 'btn btn-primary btn-sm'
+```
```

Comment:
We get the file just below. No need to show them what it _was_

```suggestion
```

Reply:
This was a common workflow in the DO workbook so I was keeping the flow consistent